lib/Support/PrettyPrinterHelpers.cpp

```diff
@@ -0,0 +1,54 @@
+//===- PrettyPrinterHelpers.cpp - Pretty printing helpers -----------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Helper classes for using PrettyPrinter.
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt/Support/PrettyPrinterHelpers.h"
+
+#include "llvm/ADT/SmallString.h"
+#include "llvm/Support/raw_ostream.h"
+
+namespace circt {
+namespace pretty {
+
+//===----------------------------------------------------------------------===//
+// Convenience builders.
+//===----------------------------------------------------------------------===//
+
+void TokenStringSaver::clear() { alloc.Reset(); }
+
+/// Add multiple non-breaking spaces as a single token.
+void detail::emitNBSP(unsigned n, llvm::function_ref<void(Token)> add) {
+  static const char spaces[] = {
+      ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ',
+      ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ',
+      ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ',
+      ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ',
+      ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ',
+      ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ',
+      ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ',
+      ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ',
+      ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ',
+  };
```

Comment:
Is there some `fill(128, ' ')` that would make this more obvious?

Reply:
https://github.com/llvm/circt/pull/4232